[Loh Jia Ming, Rayner] iP

raynerljm opened the pull request:
# Duck Task Manager 🐥

Ever wanted to manage your tasks with ~~some~~ **EXTREME** cuteness at the expense of convenience?

> “The secret of success is to be like a duck, smooth and unruffled on top, but paddling furiously underneath.” - Bill Cosby.

## Features
1. Add tasks (`todo <description>`, `deadline <description> /by <date> <time>`)
2. List tasks (`list`)
3. Mark tasks as complete (`mark <task number>`)
4. Delete tasks (`delete <task number>`)
5. Save tasks (it autosaves!)

## Types of tasks
- Todos
- Deadlines
- Events

## Future Features

- [X] Help command
- [ ] GUI

## My repository
This is a fork of the repo for a CS2103T individual project. My GitHub repo is [here](https://github.com/raynerljm/ip.git)

## Snippet
Here is a small snippet of code from the program!
```Java
public void run() {
    this.ui.printBootUp();
    boolean isExit = false;

    // Program will keep taking in new user input until terminated     while (!isExit) {
        try {
            String fullCommand = ui.readCommand();
            this.ui.printLineSeparator();
            Command command = Parser.parse(fullCommand);
            command.execute(this.tasks, this.ui, this.storage);
            isExit = command.isExit();
        } catch (DukeException e) 
             this.ui.printException(e);
        } finally {
            this.ui.printLineSeparator();
        }
    }
}
```

Commit:
Add Gradle support

raynerljm committed:
Implement level-1

raynerljm committed:
Implement level 2 commands with some comments

raynerljm committed:
Implement level-3 commands

raynerljm committed:
Implement level-4

raynerljm committed:
Add help command

raynerljm committed:
Implement A-TextUiTesting

raynerljm committed:
Refactor tasks

raynerljm committed:
Cannot add todos without description and adding a deadline or event without preposition and deadline/time means it does not show the deadline/time in brackets

raynerljm committed:
Implement level-6 delete as well as add more test cases for error handling:

raynerljm committed:
Implement the features for saving using an external file save.txt

raynerljm committed:
Force deadline and events to have a date/time

raynerljm committed:
Complete level 8

raynerljm committed:
Fix merge conflicts

raynerljm committed:
Create UI class add a run method to Duke

raynerljm committed:
Create UI class

raynerljm committed:
Add storage class and tasklist class

raynerljm committed:
Implement more OOP

Created multiple different Command classes (AddCommand, Delete, List, Exit, Help, Mark) to be used
together with the Parser class which creates these commands

raynerljm committed:
Organize the classes into packages

raynerljm committed:
Create JUnit tests for Task class

raynerljm committed:
Add meta files

raynerljm committed:
Add JavaDocs to the code

raynerljm committed:
Change done to mark

raynerljm committed:
Modify code to fit coding standards.

Examples include boolean names, boolean setters, one liner if clauses

raynerljm committed:
Implement find command

raynerljm committed:
Merge branch-A-JavaDoc into main

raynerljm committed:
Merge branch-A-CodingStandard to master

raynerljm committed:
Merge branch-Level-9 into master

raynerljm committed:
Merge add-gradle-support branch into branch-A-Gradle

raynerljm committed:
Set up gradle for build

raynerljm committed:
Merge branch-A-Gradle into master

raynerljm committed:
Install checkstyles and fix all checkstyle problems

raynerljm committed:
Merge branch-A-CheckStyle into master

raynerljm committed:
Complete part 3 for the JavaFX tutorial

raynerljm committed:
Finish part 4 of the JavaFX tutorial

Rdac0 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,167 @@
+ * Represents a command to add a new task.
+ */
+public class AddCommand extends Command {
+    /**
+     * Type of task. Can be 'todo', 'deadline', or 'event
+     */
+    private String taskType;
+    /**
+     * The description and name of the task.
+     */
+    private String taskName;
+    /**
+     * The preposition for the date time. Only applicable for deadlines and events.
+     */
+    private String taskPreposition;
+    /**
+     * The date time for the event. Only applicable for deadlines and events.
+     */
+    private LocalDateTime taskDate;
+    /**
+     * To store any exceptions that were thrown during the parsing of the command.
+     */
+    private DukeException exception;
```
I really like how thorough you are with these JavaDoc comments. 
Although, for completeness, I think the first comment (line 23) might be missing a few characters.
Shouldn't `...event` be `...event'.` ?

Rdac0 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,167 @@
+            this.exception = new DukeException("Please add a description of your todo as it cannot be empty!");
+            return;
+        }
+        //Setting up datetime
```
Shouldn't this line (and a few others) be separated by a blank line?

Rdac0 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,167 @@
+    /**
+     * Returns false as this is not an exit command.
+     *
+     * @return
```
Shouldn't you say something here? I think just a `False.` would suffice

riakhaitan replied:
agreed with @Rdac0 , otherwise it is ambiguous

Rdac0 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,38 @@
+     */
+    private String fullCommand;
+
+    Command(String fullCommand) {
```
Shouldn't constructors have JavaDoc comments too?
I'm not too sure about this.

Rdac0 reviewed (commented):
I haven't looked too deep into your code, but I did find some coding style mishaps.

raynerljm committed:
Complete Level-10 GUI

raynerljm committed:
Merge branch-Level-10

zechajw reviewed (commented):
I like the JUnit files for the Task class! They are very comprehensive

raynerljm committed:
Fix UI line length

raynerljm committed:
Add assertion to code for A-Assertions

raynerljm committed:
Remove commented unused code from Duke

raynerljm committed:
Improve code quality

Some refactoring needs to be done as the code quality can be improved.

What is being done about it
* Remove initialized variables that are not used
* Abstract out commonly used Magic literals like delimiters (;)

raynerljm committed:
Merge pull request #1 from raynerljm/branch-A-Assertions

Add assertions

raynerljm committed:
Fix merge conflicts

raynerljm committed:
Merge pull request #2 from raynerljm/branch-A-CodeQuality

Branch a code quality

raynerljm committed:
Add yml file

raynerljm committed:
Add tag functionality

Tags can help users to tag things and find them.

* Add tags to Task as an attribute.
* Add returnTags method.
* Add Tag command.

raynerljm committed:
Add save functionality to tags

raynerljm committed:
Merge pull request #3 from raynerljm/branch-C-Tagging

Branch c tagging

riakhaitan commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,167 @@
+     * @throws DukeException
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```
Love the formatting on this function, easy to understand!

riakhaitan commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,87 @@
+/**
+ * Represents a command to mark or unmark a task.
+ */
+public class MarkCommand extends Command {
```
Good job creating a separate class for commands, good use of OOP

riakhaitan commented on src/main/java/duke/function/Storage.java:
```diff
@@ -0,0 +1,121 @@
+     *
+     * @param filePath The path to the save file
+     */
+    public Storage(String filePath) {
```
Consider changing the function name as "Storage" isn't apt

raynerljm committed:
Add new user profile pictures and add bg color and padding to dialog

raynerljm committed:
Set theme jekyll-theme-hacker

raynerljm committed:
ÂUpdate user guide

raynerljm committed:
Merge

raynerljm committed:
Fix the path issues after conducting smoke tests

raynerljm committed:
Fix most automated code analysis issues

raynerljm committed:
Fix checkstyle issue
